pkg/azurediskplugin/main.go

```diff
@@ -86,38 +105,90 @@ func main() {
 }
 
 func handle() {
-	driverOptions := azuredisk.DriverOptions{
-		NodeID:                     *nodeID,
-		DriverName:                 *driverName,
-		VolumeAttachLimit:          *volumeAttachLimit,
-		EnablePerfOptimization:     *enablePerfOptimization,
-		CloudConfigSecretName:      *cloudConfigSecretName,
-		CloudConfigSecretNamespace: *cloudConfigSecretNamespace,
-		CustomUserAgent:            *customUserAgent,
-		UserAgentSuffix:            *userAgentSuffix,
-		UseCSIProxyGAInterface:     *useCSIProxyGAInterface,
-		EnableDiskOnlineResize:     *enableDiskOnlineResize,
-		AllowEmptyCloudConfig:      *allowEmptyCloudConfig,
-		EnableAsyncAttach:          *enableAsyncAttach,
-		EnableListVolumes:          *enableListVolumes,
-		EnableListSnapshots:        *enableListSnapshots,
-		SupportZone:                *supportZone,
-		GetNodeInfoFromLabels:      *getNodeInfoFromLabels,
-		EnableDiskCapacityCheck:    *enableDiskCapacityCheck,
-		VMSSCacheTTLInSeconds:      *vmssCacheTTLInSeconds,
-		VMType:                     *vmType,
-		RestClientQPS:              *kubeClientQPS,
-	}
-	driver := azuredisk.NewDriver(&driverOptions)
-	if driver == nil {
-		klog.Fatalln("Failed to initialize azuredisk CSI Driver")
+	var driverOptions azuredisk.DriverOptions
+	if *driverConfigPath != "" {
```

Comment:
Consider creating a `getDriverConfig` function which initializes a new `AzDriverConfiguration` object either from the file (if specified) or command-line parameters.